packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1135,10 +1141,22 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         if (shouldRetryRequest) {
             this.stats.errorTrackerRetry.add(error);
 
-            await this._tagUserHandlerError(() => this.errorHandler?.(this._augmentContextWithDeprecatedError(crawlingContext, error), error));
+            if (error instanceof SessionError) {
+                if ((request.sessionRotationCount ?? 0) > 10) {
```

Comment:
Should the maximum number be parametrizable?

Reply:
Maybe it would be enough to have this in the `Configuration` class?

Reply:
Huh, I've always assumed the `Configuration` class to be something "above" the scraping logic (logging levels, output directory etc.), IMO this "perseverance" of the crawler is something related to the current run / scraped website... 

I'm undecided, it feels like adding it to the `CrawlerOptions` would make it a bit confusing (`maxRetries` / `maxSessionRotations`), but maybe those are distinct enough 🤔